### DISCUSSION TITLE
How to share insert-node in choices?

### DISCUSSION THREAD

**Discussion by fovj** [L3MON4D3/LuaSnip#542] [Q&A] [answered]
```lua
  s("dd", {
    c(1, {
      fmt("{}", { i(1, "content") }),
      fmt("({})", { i(1, "content") }),
      fmt("[{}]", { i(1, "content") }),
    }),
  }),
```
![ezgif-2-b5ef13c8e4](https://user-images.githubusercontent.com/4456413/185730599-b08abbd0-39a9-418b-a87a-b94becbc4952.gif)

**Comment by L3MON4D3** [comment, marked as answer]
Ah, you can use [`restoreNode`](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#restorenode) to share nodes.

**Reply by fovj** [reply]
THAT'S AWESOME!

**Reply by L3MON4D3** [reply]
Now that's the kind of reactions I'm hoping for xD
